File: app/src/main/java/com/wavesplatform/wallet/v2/ui/custom/FadeInWithoutDelayAnimator.java
/*
 * Created by Eduard Zaydel on 1/4/2019
 * Copyright © 2019 Waves Platform. All rights reserved.
 */

package com.wavesplatform.wallet.v2.ui.custom;

import android.support.v4.view.ViewCompat;
import android.support.v7.widget.RecyclerView;
import android.view.animation.Interpolator;

import jp.wasabeef.recyclerview.animators.BaseItemAnimator;

public class FadeInWithoutDelayAnimator extends BaseItemAnimator {

    public FadeInWithoutDelayAnimator() {
    }

  public FadeInWithoutDelayAnimator(Interpolator interpolator) {
    mInterpolator = interpolator;
  }

  @Override protected void animateRemoveImpl(final RecyclerView.ViewHolder holder) {
    ViewCompat.animate(holder.itemView)
        .alpha(0)
        .setDuration(getRemoveDuration())
        .setInterpolator(mInterpolator)
        .setListener(new DefaultRemoveVpaListener(holder))
        .start();
  }

  @Override protected void preAnimateAddImpl(RecyclerView.ViewHolder holder) {
    ViewCompat.setAlpha(holder.itemView, 0);
  }

  @Override protected void animateAddImpl(final RecyclerView.ViewHolder holder) {
    ViewCompat.animate(holder.itemView)
        .alpha(1)
        .setDuration(getAddDuration())
        .setInterpolator(mInterpolator)
        .setListener(new DefaultAddVpaListener(holder))
        .start();
  }
}

File: app/src/main/java/com/wavesplatform/wallet/v2/ui/custom/FadeInWithoutDelayAnimator.kt
/*
 * Created by Eduard Zaydel on 1/4/2019
 * Copyright © 2019 Waves Platform. All rights reserved.
 */

package com.wavesplatform.wallet.v2.ui.custom

import androidx.core.view.ViewCompat
import androidx.recyclerview.widget.RecyclerView
import android.view.animation.Interpolator

import jp.wasabeef.recyclerview.animators.BaseItemAnimator

class FadeInWithoutDelayAnimator : BaseItemAnimator {

    var animationEnable = true

    constructor()

    constructor(interpolator: Interpolator) {
        mInterpolator = interpolator
    }

    override fun animateRemoveImpl(holder: androidx.recyclerview.widget.RecyclerView.ViewHolder) {
        if (animationEnable) {
            ViewCompat.animate(holder.itemView)
                    .alpha(0f)
                    .setDuration(removeDuration)
                    .setInterpolator(mInterpolator)
                    .setListener(DefaultRemoveVpaListener(holder))
                    .start()
        }
    }

    override fun preAnimateAddImpl(holder: androidx.recyclerview.widget.RecyclerView.ViewHolder?) {
        if (animationEnable) {
            ViewCompat.setAlpha(holder!!.itemView, 0f)
        }
    }

    override fun animateAddImpl(holder: androidx.recyclerview.widget.RecyclerView.ViewHolder) {
        if (animationEnable) {
            ViewCompat.animate(holder.itemView)
                    .alpha(1f)
                    .setDuration(addDuration)
                    .setInterpolator(mInterpolator)
                    .setListener(DefaultAddVpaListener(holder))
                    .start()
        }
    }
}
